### PR TITLE
Forms HTML5

### PR DESCRIPTION
**Pull request by CrazyPike**
### HTML5 Form

[Demo](https://crazypike.github.io/frontend-2019-homeworks/submissions/CrazyPike/formss/index.html) |
[Code base](https://github.com/CrazyPike/frontend-2019-homeworks/tree/html5form/submissions/CrazyPike/formss)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/CrazyPike/formss/index.html [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+<html lang="ru">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Главная страница</title>
+    <link href="main.css" rel="stylesheet">
+  </head>
+  <body class="page__common">
+    <nav class="menu">
+      <button class="menu__button"><span class="menu__button-text"><img class="menu__button-img" src="img/arrow_back.png" alt="Arrow back">GO BACK</span></button><a href="#"><img class="menu__logo" src="img/logo.png" alt="Here is a logo"></a>
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.html]: Add styles on `:hover`, `:active`, and `:focus` state.

File: submissions/CrazyPike/formss/index.html [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+      <button class="menu__button"><span class="menu__button-text"><img class="menu__button-img" src="img/icon_profile.png" alt="Log in">SIGN IN</span></button>
+    </nav>
+    <main class="form">
+      <form id="html5form"></form>
+      <section class="form__common-info">
+        <div class="quote"><img class="form__icons" src="img/icon_order.png" alt="Order icon">
+          <div class="form__header-text">YOUR ORDER</div>
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.html]: Don't use text in uppercase as content. Use CSS for this. When you change text your visual representation should be the same and independent of text content or text case inside HTML.

File: submissions/CrazyPike/formss/index.html [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            </thead>
+            <tr>
+              <td> 2 documents</td>
+              <td> 2 languages</td>
+              <td> $ 696.06</td>
+              <td> 48 hours</td>
+            </tr>
+          </table>
+          <div class="quote__buttons">
+            <button class="menu__button quote__button-place-order"><span class="menu__button-text">PLACE ORDER</span></button>
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.html]: Add the attribute `type` to the button.

**Comment by CrazyPike** [comment]
@A-Ostrovnyy i've pushed the new version of code. please review

File: submissions/CrazyPike/formss/index.pug [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+                div.contacts
+                    img.form__icons(src='img/icon_help.png', alt='Help icon')
+                    div.form__header-text help center
+                    div.contacts__desc
+                        | If you need assistance while placing your
+                        | order, contact one of our beloved Project
+                        | Manager by phone or email
+                    div.contacts__details
+                        a(href='tel:+9710555555') +971 (0) 555 555
+                        span help@qtest.com
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Use same way(`<a>`) for the e-mail as for phone number.

File: submissions/CrazyPike/formss/index.pug [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+            section.section-translate
+                div.section-translate__desc select what you want to translate
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Use headings `h1-h6` for this text.

File: submissions/CrazyPike/formss/index.pug [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+                    div.section-translate__text__center
+                    hr.section-translate__horizontal-line
+                div.section-translate__options
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Radio-buttons inside this section must be required. Also, one of them must be checked by default.

File: submissions/CrazyPike/formss/index.pug [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+                    input.section-translate__input__hidden(type='radio', name='type-for-translate',id='option-web',
+                    value='web')
+                    label.section-translate__buttons(for="option-web")
+                        img(src='img/icon_website.png')
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Add and fill `alt` attribute of `img` tag.

File: submissions/CrazyPike/formss/index.pug [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+                        img(src='img/icon_documents.png')
+                        span Documents
+                    input.section-translate__input__hidden(type='radio', name='type-for-translate',id='option-social',
+                        value='web')
+                    label.section-translate__buttons(for="option-social")
+                        img(src='img/icon_social.png')
+                        span Social Accounts
+            section.form__section-upload
+                div.section__ordinal-number 1
+                div.section__header Upload your documents
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Use headings `h1-h6` for this text.

File: submissions/CrazyPike/formss/index.pug [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+                            td.section-upload__table-col-4
+                                button.form__delete-button(type='button')
+                                    img(src="img/icon_trash.png", alt='Trash icon')
+
+            section.form__section-languages
+                div.section__ordinal-number 2
+                div.section__header Choose your languages pairings
+                div.section-languages__content-container
+                    div.section-languages__content
+                        div FROM
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Avoid using `div` without classname/id. Also, don't write text in uppercase.

File: submissions/CrazyPike/formss/index.pug [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+                        select.section-languages__select
+                            option Arabic
+                            option Slavic
+                        div
+                        div
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: WTF?

**Comment by A-Ostrovnyy** [comment]
Please update your demo.

File: submissions/CrazyPike/formss/index.pug [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+            section.form__section-tone-urgent
+                div.section-tone
+                    div.section__ordinal-number 3
+                    div.section__header Select your desired tone
+                    div.section-tone__container
+                        input.section-tone__buttons__hidden(type='radio', name='tone',
+                        id='option-friendly',value='friwndly')
+                        label.section-tone__buttons(for="option-friendly", tabindex="0")
+                            div
+                                img(src='img/icon_friendly.png')
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Add and fill `alt` attribute of `img` tag.

File: submissions/CrazyPike/formss/index.pug [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+                                button.form__delete-button(type='button')
+                                    img(src="img/icon_trash.png", alt='Trash icon')
+
+            section.form__section-languages
+                div.section__ordinal-number 2
+                div.section__header Choose your languages pairings
+                div.section-languages__content-container
+                    div.section-languages__content
+                        div FROM
+                        select.section-languages__select
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This input is required.

File: submissions/CrazyPike/formss/index.pug [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+                            option Arabic
+                            option Slavic
+                        div
+                        div
+                        div TO
+                        select.section-languages__select
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This input is required too.

File: submissions/CrazyPike/formss/index.pug [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+                                option Polish
+                            button.form__delete-button(type='button')
+                                img(src="img/icon_trash.png", alt='Trash icon')
+                        button.section-languages__button-add(type='button') + ADD
+            section.form__section-tone-urgent
+                div.section-tone
+                    div.section__ordinal-number 3
+                    div.section__header Select your desired tone
+                    div.section-tone__container
+                        input.section-tone__buttons__hidden(type='radio', name='tone',
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Radio-buttons inside this section must be required. Also, one of them must be checked by default.

File: submissions/CrazyPike/formss/index.pug [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+                                span Media
+
+                div.section-urgent
+                    div.section__ordinal-number 4
+                    div.section__header Tell us how urgent this job is
+                    div.section-urgent__range
+                        div.section-urgent__range-end
+                            div.input-range-end
+                            div.input-range-end
+                        input(type="range", min="0", max="2", value="1" id="form__range")
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This input is required.

File: submissions/CrazyPike/formss/index.pug [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+                        div.section-urgent__range-label
+                            span i got time
+                            span average
+                            span yesterday
+                    div.section-urgent__description
+                        img.form__icons(src="img/icon_info.png",alt='')
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: `alt` attribute is empty.

File: submissions/CrazyPike/formss/index.pug [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+                            | take a reasonable amount of time perfecting your translation
+            section.form-payment
+                div.section__ordinal-number 5
+                div.section__header Sign up and choose a payment method
+                div.form-payment__signin-text Already have an account with us?
+                    a(href="#") Sign In
+                div.form-payment__input-name
+                    div.form-payment__input
+                        span full name
+                        input(type='text', placeholder='Full name', pattern='[0-9A-Za-z. -]+',
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This input is required.

File: submissions/CrazyPike/formss/index.pug [hunk 18]
```diff
@@ -0,0 +1,255 @@
...
+                            span With the AVERAGE timeline set up, our expert translator can
+                            | take a reasonable amount of time perfecting your translation
+            section.form-payment
+                div.section__ordinal-number 5
+                div.section__header Sign up and choose a payment method
+                div.form-payment__signin-text Already have an account with us?
+                    a(href="#") Sign In
+                div.form-payment__input-name
+                    div.form-payment__input
+                        span full name
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This text should be a `label`.

File: submissions/CrazyPike/formss/index.pug [hunk 19]
```diff
@@ -0,0 +1,255 @@
...
+                        input(type='text', placeholder='Full name', pattern='[0-9A-Za-z. -]+',
+                        title='Enter the name')
+                div.form-payment__input-email-pass
+                    div.form-payment__input
+                        span email address
+                        input(type='email', placeholder='Email',
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This input is required.

File: submissions/CrazyPike/formss/index.pug [hunk 20]
```diff
@@ -0,0 +1,255 @@
...
+                        pattern='[0-9A-Za-z.-]+[@][0-9A-Za-z.-]+[.][0-9A-Za-z.-]+',
+                        title='Email address')
+                    div.form-payment__input
+                        span password
+                        input(type='password', minlength='6', placeholder='Enter password..',
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This input is required.

File: submissions/CrazyPike/formss/index.pug [hunk 21]
```diff
@@ -0,0 +1,255 @@
...
+                        title='Enter password')
+                div.form-payment__option-container
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Radio-buttons inside this section must be required. Also, one of them must be checked by default.

File: submissions/CrazyPike/formss/index.pug [hunk 22]
```diff
@@ -0,0 +1,255 @@
...
+                    input.section-payment__input__hidden(type='radio', name='payment',id='option-card',value='card')
+                    label.form-payment__option(for="option-card", tabindex="0")
+                        img(src='img/icon_cards.png')
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Add padding and fill attribute `alt`

File: submissions/CrazyPike/formss/index.pug [hunk 23]
```diff
@@ -0,0 +1,255 @@
...
+                    input.section-payment__input__hidden(type='radio', name='payment',id='option-paypal',value='paypal')
+                    label.form-payment__option(for="option-paypal", tabindex="0")
+                        img(src='img/icon_paypal.png')
+                    input.section-payment__input__hidden(type='radio', name='payment',id='option-amazon',value='amazon')
+                    label.form-payment__option(for="option-amazon")
+                        img(src='img/icon_amazon.png')
+                div.form-payment__card-info
+                    div.form-payment__card-input
+                        span card number
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This text should be a `label`.

File: submissions/CrazyPike/formss/index.pug [hunk 24]
```diff
@@ -0,0 +1,255 @@
...
+                        title='Format MM/YY')
+                    div.form-payment__card-input
+                        span card cvv
+                        input.form-payment__card-cvv(type='password', pattern='[0-9]{3}',placeholder='CVV', minlength=3,
+                        maxlength=3,title='CVV mush have 3 digits')
+                        span
+            section.form-submit
+                div
+                    input(type='checkbox',id='TC-label')
+                    label(for='TC-label') I agree to the terms & Conditions
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Term & conditions must be a link.

File: submissions/CrazyPike/formss/index.pug [hunk 25]
```diff
@@ -0,0 +1,255 @@
...
+                button.menu__button.quote__button-place-order(type='button')
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: `type="submit"`

File: submissions/CrazyPike/formss/index.pug [hunk 26]
```diff
@@ -0,0 +1,255 @@
...
+        footer.footer
+            div.footer__container
+                section.footer__item
+                    img.footer__item-icon(src='img/icon_heart.png', alt='heart icon')
+                    span.footer__item-header what we do
+                    span.footer__item-content We provide a full range of language
+                        | services including content development,
+                        | document, website and social media
+                        | accounts translation
+                section.footer__item
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Use the tag `address` for information like this.

File: submissions/CrazyPike/formss/index.pug [hunk 27]
```diff
@@ -0,0 +1,255 @@
...
+                    img.footer__item-icon(src='img/icon_compass.png', alt='Compass icon')
+                    span.footer__item-header our main office
+                    span.footer__item-content Dubai, UAE
+                        br
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Avoid using `br` tag

File: submissions/CrazyPike/formss/index.pug [hunk 28]
```diff
@@ -0,0 +1,255 @@
...
+                        span P.O. Box 832472
+                        br
+                        span Phone
+                            a(href='tel:+9710555555') +971 (0) 44278128
+                section.footer__item
+                    img.footer__item-icon(src='img/icon_network.png', alt='Network icon')
+                    span.footer__item-header connect with us
+                    span.footer__item-content
+                            div.footer__social-item
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This must be links.

File: submissions/CrazyPike/formss/index.pug (hunk 10 above)
Comment (CrazyPike) [reply submissions/CrazyPike/formss/index.pug]: added the class. I've created the table , and didn't want to create an additional grid layout. So i've  created just empty class for empty divs.

File: submissions/CrazyPike/formss/index.pug (hunk 4 above)
Comment (CrazyPike) [reply submissions/CrazyPike/formss/index.pug]: done

File: submissions/CrazyPike/formss/index.pug (hunk 5 above)
Comment (CrazyPike) [reply submissions/CrazyPike/formss/index.pug]: done

File: submissions/CrazyPike/formss/index.pug (hunk 6 above)
Comment (CrazyPike) [reply submissions/CrazyPike/formss/index.pug]: done

File: submissions/CrazyPike/formss/index.pug (hunk 7 above)
Comment (CrazyPike) [reply submissions/CrazyPike/formss/index.pug]: done

File: submissions/CrazyPike/formss/index.pug (hunk 8 above)
Comment (CrazyPike) [reply submissions/CrazyPike/formss/index.pug]: done

File: submissions/CrazyPike/formss/index.pug (hunk 10 above)
Comment (A-Ostrovnyy) [reply submissions/CrazyPike/formss/index.pug]: It's generated into three empty `div` without classes.

**Comment by CrazyPike** [comment]
done

File: submissions/CrazyPike/formss/index.pug [hunk 29]
```diff
@@ -41,39 +41,39 @@ html(lang="ru")
...
+                        a.page__links(href='mailto:help@qtest.com') help@qtest.com
             section.section-translate
-                div.section-translate__desc select what you want to translate
+                h2.section-translate__desc select what you want to translate
                     div.section-translate__text__center
                     hr.section-translate__horizontal-line
                 div.section-translate__options
                     input.section-translate__input__hidden(type='radio', name='type-for-translate',id='option-web',
-                    value='web')
+                    value='web' required="required" checked='checked')
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: You just needed to add one word: `required`  instead of `required="required"`.

Comment (CrazyPike) [reply submissions/CrazyPike/formss/index.pug]: fixed,

File: submissions/CrazyPike/formss/index.pug [hunk 30]
```diff
@@ -223,31 +224,29 @@ html(lang="ru")
             div.footer__container
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Remove extra margin in the footer. [Image](https://imgur.com/a/tilNyQe)

File: submissions/CrazyPike/formss/index.pug [hunk 31]
```diff
@@ -106,21 +106,21 @@ html(lang="ru")
...
             section.form__section-languages
                 div.section__ordinal-number 2
-                div.section__header Choose your languages pairings
+                h2.section__header Choose your languages pairings
                 div.section-languages__content-container
                     div.section-languages__content
-                        div FROM
-                        select.section-languages__select
+                        label.page__uppercase from
+                        select.section-languages__select(required='required')
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Add a vertical line in every `select`. [Image](https://imgur.com/a/42NUWbC)

File: submissions/CrazyPike/formss/index.pug [hunk 32]
```diff
@@ -129,92 +129,93 @@ html(lang="ru")
...
                 div.form-payment__signin-text Already have an account with us?
-                    a(href="#") Sign In
+                    a.page__links(href="#") Sign In
                 div.form-payment__input-name
                     div.form-payment__input
-                        span full name
-                        input(type='text', placeholder='Full name', pattern='[0-9A-Za-z. -]+',
-                        title='Enter the name')
+                        label(for='person_name') full name
+                        input(type='text', placeholder='Full name', pattern='[0-9A-Za-z. -]+', id = 'person_name',
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Add paddings to every text/date/number input.
Also, make `font-size` bigger.

File: submissions/CrazyPike/formss/index.pug [hunk 33]
```diff
@@ -129,92 +129,93 @@ html(lang="ru")
...
                         div.section-urgent__description-text
                             span With the AVERAGE timeline set up, our expert translator can
                             | take a reasonable amount of time perfecting your translation
             section.form-payment
                 div.section__ordinal-number 5
-                div.section__header Sign up and choose a payment method
+                h2.section__header Sign up and choose a payment method
                 div.form-payment__signin-text Already have an account with us?
-                    a(href="#") Sign In
+                    a.page__links(href="#") Sign In
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Change all links to the corresponding color.

File: submissions/CrazyPike/formss/index.pug (hunk 30 above)
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: Fix footer background on the mobile view. [Image](https://imgur.com/a/7sdjpOU)

File: submissions/CrazyPike/formss/index.pug [hunk 34]
```diff
@@ -223,31 +224,29 @@ html(lang="ru")
             div.footer__container
                 section.footer__item
                     img.footer__item-icon(src='img/icon_heart.png', alt='heart icon')
-                    span.footer__item-header what we do
+                    h4.footer__item-header what we do
                     span.footer__item-content We provide a full range of language
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This text must be `italic`.

File: submissions/CrazyPike/formss/index.pug [hunk 35]
```diff
@@ -223,31 +224,29 @@ html(lang="ru")
...
                         | services including content development,
                         | document, website and social media
                         | accounts translation
                 section.footer__item
                     img.footer__item-icon(src='img/icon_compass.png', alt='Compass icon')
-                    span.footer__item-header our main office
-                    span.footer__item-content Dubai, UAE
-                        br
+                    h4.footer__item-header our main office
+                    address.footer__item-content Dubai, UAE
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/formss/index.pug]: This text, not `italic`.

**Pull request merged**